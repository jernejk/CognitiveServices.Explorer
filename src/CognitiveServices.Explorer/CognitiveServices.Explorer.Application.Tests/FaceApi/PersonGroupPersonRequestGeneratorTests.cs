using CognitiveServices.Explorer.Application.FaceApi;
using FluentAssertions;
using Snapshooter.Xunit;
using System;
using Xunit;

namespace CognitiveServices.Explorer.Application.Tests.FaceApi
{
    public class PersonGroupPersonRequestGeneratorTests
    {
        [Fact]
        public void ShouldGenerateCreate()
        {
            PersonGroupPersonRequestGenerator
                .Create("default", "User", @"{""test"":""data""}")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateUpdate()
        {
            PersonGroupPersonRequestGenerator
                .Update("default", Guid.Empty.ToString(), "User2")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateDelete()
        {
            PersonGroupPersonRequestGenerator
                .Delete("default", Guid.Empty.ToString())
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateGet()
        {
            PersonGroupPersonRequestGenerator
                .Get("default", Guid.Empty.ToString())
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateList()
        {
            PersonGroupPersonRequestGenerator
                .List("default")
                .Should()
                .MatchSnapshot();
        }
    }
}
