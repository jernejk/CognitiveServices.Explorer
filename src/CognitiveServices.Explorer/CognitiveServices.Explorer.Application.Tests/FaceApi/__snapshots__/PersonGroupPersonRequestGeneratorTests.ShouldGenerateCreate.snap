{
  "RequestName": null,
  "Headers": {},
  "RelativePath": "face/v1.0/persongroups/default/persons",
  "Queries": null,
  "HttpMethod": "POST",
  "Body": "{\"name\":\"User\",\"userData\":\"{\\u0022test\\u0022:\\u0022data\\u0022}\"}",
  "BinaryContent": null,
  "ContentType": "application/json",
  "ExpectedHtppCode": 200,
  "TokenHeaderName": "Ocp-Apim-Subscription-Key",
  "Cost": {
    "ServiceName": "FaceApi",
    "Cost": 1.0,
    "Unit": "tr"
  },
  "CognitiveServiceDoc": "https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f3039523c",
  "AbsoluteUrl": null
}
